aws-gitlab-main.yml

```diff
@@ -0,0 +1,214 @@
+
+      #ROUTE TABLES
+      gitlab-public-RouteTable:
+        Type: 'AWS::EC2::RouteTable' 
+        Properties:
+          VpcId: !Ref gitlabVPC
+      
+      gitlab-public-Route:
+        Type: 'AWS::EC2::Route'
+        DependsOn: gitlab-GatewayAttachment
+        Properties:
+          RouteTableId: !Ref gitlab-public-RouteTable
+          DestinationCidrBlock: 0.0.0.0/0
+          GatewayId: !Ref gitlab-gateway
+      
+      gitlab-public-10.0.0.0-RouteTableAssociation:
+        Type: 'AWS::EC2::SubnetRouteTableAssociation' 
+        Properties:
+          SubnetId: !Ref gitlab-public-10.0.0.0
+          RouteTableId: !Ref gitlab-public-RouteTable
+
+      gitlab-public-10.0.2.0-RouteTableAssociation:
+        Type: 'AWS::EC2::SubnetRouteTableAssociation' 
+        Properties:
+          SubnetId: !Ref gitlab-public-10.0.2.0
+          RouteTableId: !Ref gitlab-public-RouteTable
+
+      gitlab-private-RouteTable-a:
+        Type: 'AWS::EC2::RouteTable' 
+        Properties:
+          VpcId: !Ref gitlabVPC
+      
+      gitlab-private-Route-a:
+        Type: 'AWS::EC2::Route' 
+        Properties:
+          RouteTableId: !Ref gitlab-private-RouteTable-a
+          DestinationCidrBlock: 0.0.0.0/0
+      
+      gitlab-private-10.0.1.0-RouteTableAssociation:
+        Type: 'AWS::EC2::SubnetRouteTableAssociation' 
+        Properties:
+          SubnetId: !Ref gitlab-private-10.0.1.0
+          RouteTableId: !Ref gitlab-private-RouteTable-a 
+
+      gitlab-private-RouteTable-b:
+            Type: 'AWS::EC2::RouteTable' 
+            Properties:
+              VpcId: !Ref gitlabVPC
+          
+      gitlab-private-Route-b:
+        Type: 'AWS::EC2::Route' 
+        Properties:
+          RouteTableId: !Ref gitlab-private-RouteTable-b
+          DestinationCidrBlock: 0.0.0.0/0
+
+      gitlab-private-10.0.3.0-RouteTableAssociation:
+        Type: 'AWS::EC2::SubnetRouteTableAssociation' 
+        Properties:
+          SubnetId: !Ref gitlab-private-10.0.3.0
+          RouteTableId: !Ref gitlab-private-RouteTable-b
+
+      #LOAD BALANCER
+      gitlab-LoadBalancer:
+        Type: 'AWS::ElasticLoadBalancing::LoadBalancer'
+        Properties: 
+          Listeners: 
+          - LoadBalancerPort: '80'
+            InstancePort: '80'
+            Protocol: HTTP
+          - LoadBalancerPort: '443'
+            InstancePort: '443'
+            Protocol: HTTPS
+          - LoadBalancerPort: '22'
+            InstancePort: '22'
+            Protocol: TCP
+          HealthCheck: #not sure about this section
+            Target: HTTP:80/
+            #HealthyThreshold: '3'
+            #UnhealthyThreshold: '5'
+            #Interval: '30'
+            #Timeout: '5'
+          #Req:"Ping Protocol (HTTP), Ping Port (80), Ping Path (/users/sign_in)"
+          LoadbalancerName: gitlab-LoadBalancer
+          Policies:
+            #??? 
+          Scheme: internet-facing #?
+          Security Groups: #not sure about this section
```

Comment:
Again, you cannot create resources inside resources. Tbh never even tried or seen this lol.

Create security groups separate then !Ref or !GetAtt as needed